File: environment/env_v0_3x/dirs.go
package env_v0_3x

import (
	"os"
	"path/filepath"
)

func rootDir() (string, error) {
	home, err := os.UserHomeDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(home, ".zrok"), nil
}

func metadataFile() (string, error) {
	zrd, err := rootDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(zrd, "metadata.json"), nil
}

func configFile() (string, error) {
	zrd, err := rootDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(zrd, "config.json"), nil
}

func environmentFile() (string, error) {
	zrd, err := rootDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(zrd, "environment.json"), nil
}

File: sdk/nodejs/sdk/src/environment/dirs.ts
import { homedir } from "os"
import { join } from "node:path"

export function rootDir(): string {
    return join(homedir(), ".zrok")
}

export function metadataFile(): string {
    return join(rootDir(), "metadata.json")
}

export function configFile(): string {
    return join(rootDir(), "config.json")
}

export function environmentFile(): string {
    return join(rootDir(), "environment.json")
}

export function identitiesDir(): string {
    return join(rootDir(), "identities")
}

export function identityFile(name: string): string {
    return join(identitiesDir(), name + ".json")
}